requirements.txt

```diff
@@ -0,0 +1,41 @@
+aiohttp==3.7.4.post0
+aioredis==2.0.1
+async-timeout==3.0.1
+attrs==21.2.0
+autopep8==1.6.0
+backcall==0.2.0
+chardet==4.0.0
+debugpy==1.5.1
+decorator==5.1.1
+Deprecated==1.2.13
+discord==1.7.3
+discord.py==1.7.3
+dynaconf==3.1.7
+entrypoints==0.3
+idna==3.3
+ipykernel==6.6.1
+ipython==7.31.0
+jedi==0.18.1
+jupyter-client==7.1.0
+jupyter-core==4.9.1
+matplotlib-inline==0.1.3
+multidict==5.2.0
+nest-asyncio==1.5.4
+parso==0.8.3
+pexpect==4.8.0
+pickleshare==0.7.5
+Pillow==8.4.0
```

Comment:
I was having issue building the docker container due to Pillow being required in this file. I see earlier in the dockerfile that Pillow is installed along side some virtual dependencies. That installation worked, but then my guess, based on the errors I was seeing, is that the virtual dependencies went away by the time these requirements started getting installed, and so the Pillow install from this file failed for me. I went ahead and removed this Pillow requirement locally, and the build started working.

Reply:
Hmm... I'll take a look. The Dockerfile was set up that way to limit the size of the resulting image. What system are you building on?

Reply:
Windows

Reply:
I can't reproduce this on Linux (pop_OS). I just reset my venv, wiped my local docker images, and ran `docker-compose up -d --build` and it built fine. When exactly did it fail on your end? I wonder if in the hard-coded Pillow install it's pulling a different version than that declared in requirements.txt...

That said, it's not great how the Pillow install is hard-coded in Dockerfile. Can you try it again, restoring `requirements.txt` and swapping this in for your Dockerfile:

```
# syntax=docker/dockerfile:1

FROM python:3.9-alpine

WORKDIR /srv

COPY requirements.txt requirements.txt

RUN apk update \
    && apk add --virtual build-deps gcc python3-dev musl-dev \
    && apk add jpeg-dev zlib-dev libjpeg freetype-dev \
    && pip3 install -r requirements.txt \
    && apk del build-deps
```

Reply:
New Dockerfile worked like a charm.

Old Dockerfile failed while building the wheel for Pillow.

Here's the original failure output:
```
...
#11 34.06   Building wheel for Pillow (setup.py): started
#11 34.41   Building wheel for Pillow (setup.py): finished with status 'error'
#11 34.41   ERROR: Command errored out with exit status 1:
#11 34.41    command: /usr/local/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-uznehizv/pillow_b1093849bbb14b3186f63f19448e2a4e/setup.py'"'"'; __file__='"'"'/tmp/pip-install-uznehizv/pi
llow_b1093849bbb14b3186f63f19448e2a4e/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"
'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-tql45hky
#11 34.41        cwd: /tmp/pip-install-uznehizv/pillow_b1093849bbb14b3186f63f19448e2a4e/
...
#11 34.41   error: command 'gcc' failed: No such file or directory
#11 34.41   ----------------------------------------
#11 34.41   ERROR: Failed building wheel for Pillow
...
```

Reply:
Great, just pushed the new one.